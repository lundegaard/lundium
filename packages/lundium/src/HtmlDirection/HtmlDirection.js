import React from 'react';
import Helmet from 'react-helmet';

import { useTheme } from '../ThemeProvider';

/**
 * Ensures right direction attribute on HTML element.
 */
const HtmlDirection = () => {
	const { isRTL } = useTheme();
	const htmlAttributes = { dir: isRTL ? 'rtl' : 'ltr' };
	return <Helmet htmlAttributes={htmlAttributes} />;
};

export default HtmlDirection;
