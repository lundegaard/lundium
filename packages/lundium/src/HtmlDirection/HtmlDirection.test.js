import React from 'react';
import { shallow } from 'enzyme';

import HtmlDirection from './HtmlDirection';

describe('<HtmlDirection />', () => {
	const requiredProps = {};

	it('matches snapshot when shallowly rendered', () => {
		const wrapper = shallow(<HtmlDirection {...requiredProps} />);
		expect(wrapper).toMatchSnapshot();
	});
});
