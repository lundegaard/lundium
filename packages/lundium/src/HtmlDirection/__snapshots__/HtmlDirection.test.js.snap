// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HtmlDirection /> matches snapshot when shallowly rendered 1`] = `
<HelmetWrapper
  defer={true}
  encodeSpecialCharacters={true}
  htmlAttributes={
    Object {
      "dir": "ltr",
    }
  }
/>
`;
